src/wallet/mod.rs

```diff
@@ -1743,6 +1740,12 @@ where
         self.database.borrow()
     }
 
+    #[cfg(test)]
+    /// Return an mutable reference to the internal database
+    pub(crate) fn database_mut(&self) -> impl std::ops::DerefMut<Target = D> + '_ {
+        self.database.borrow_mut()
+    }
+
```

Comment:
I don't understand exactly why this is necessary

Reply:
Me neither.. But when I tried to directly use `self.database.borrow_mut()` in the function calls the compiler got angry and threw lots of problems that I couldn't subvert, and the easiest thing I found was to have this redundant function.. This does bother me, as we should not need something like this.. But for time being this works.. And I will try to remove it again and see if it can work.. But would prefer to do it in a separate PR..

Reply:
Another problem faced is previously the `populate_test_db` macro was the the database module.. Which made accessing database easy.. But now as it has been taken out into `testutils::helper` we cannot just call `wallet.db` as that's private variable and only exposed via the `database()` api.. Which only gives immutable access.. So thats the major reason I had to make a mutable version of the same API.. But only for testing situations..

Reply:
We can make the `Wallet::database` field `pub(crate)`.